src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,38 @@
      * Андрей - 600
      * София - 900</p>
      */
+
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        final LocalDate dateToDate = LocalDate.parse(dateTo, FORMATTER);
+        final LocalDate dateFromDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate dateOfLine;
+        int money;
```

Comment:
since we use these variables only inside of loop, let's declare them there

Reply:
done